### PR TITLE
Added button to export voting history into JSON

### PR DESCRIPTION
**Pull request by ben-kaufman**
Fix: #2477 

Added a button to export the voing history under DAO -> Governance -> Vote Results:
screen shot 2019-03-06 at 18 13 36

Attached here is the JSON file which was generated for me so it's possible to compare when you generate locally (zipped because GitHub doesn't support `.json`). 

Note: I'm not confident with the way I added the export button to the UI.

[voteResultsHistory.json.zip](https://github.com/bisq-network/bisq/files/2937159/voteResultsHistory.json.zip)

### CODE REVIEW

File: desktop/src/main/java/bisq/desktop/main/dao/governance/result/VoteResultView.fxml [hunk 1]
```diff
@@ -29,4 +31,7 @@
     <columnConstraints>
         <ColumnConstraints percentWidth="100"/>
     </columnConstraints>
+    <HBox GridPane.rowIndex="8" GridPane.columnSpan="3" alignment="BOTTOM_RIGHT">
```
Comment (ManfredKarrer) [inline desktop/src/main/java/bisq/desktop/main/dao/governance/result/VoteResultView.fxml]: We do not use FXML anymore beside of the containers (one day we want to get rid completely of it) as it has bad performance and is too limited for more dynamic content. There are still relicts but new code we want to have in Java only. See FormBuilder as out Util for UI components.

File: desktop/src/main/java/bisq/desktop/main/dao/governance/result/VoteResultView.java [hunk 2]
```diff
@@ -105,6 +112,7 @@
     private final Preferences preferences;
     private final BsqFormatter bsqFormatter;
     private final Navigation navigation;
+    public AutoTooltipButton exportButton;
```
Comment (ManfredKarrer) [inline desktop/src/main/java/bisq/desktop/main/dao/governance/result/VoteResultView.java]: Would be private once removed from FXML

File: desktop/src/main/java/bisq/desktop/main/dao/governance/result/VoteResultView.java [hunk 3]
```diff
@@ -157,9 +165,11 @@ public VoteResultView(DaoFacade daoFacade,
     @Override
     public void initialize() {
         gridRow = phasesView.addGroup(root, gridRow);
+
         selectedVoteResultListItemListener = (observable, oldValue, newValue) -> onResultsListItemSelected(newValue);
 
         createCyclesTable();
+        exportButton.updateText(Res.get("shared.exportJSON"));
```
Comment (ManfredKarrer) [inline desktop/src/main/java/bisq/desktop/main/dao/governance/result/VoteResultView.java]: Here would be the code or creating the Button. 
E.g. `button = FormBuilder.addButtonAfterGroup( gridPane,  rowIndex,  title);`

File: desktop/src/main/java/bisq/desktop/main/dao/governance/result/VoteResultView.java [hunk 4]
```diff
@@ -172,6 +182,57 @@ protected void activate() {
         cyclesTableView.getSelectionModel().selectedItemProperty().addListener(selectedVoteResultListItemListener);
 
         fillCycleList();
+        exportButton.setOnAction(event -> {
```
Comment (ManfredKarrer) [inline desktop/src/main/java/bisq/desktop/main/dao/governance/result/VoteResultView.java:198]: I would prefer to delegate that to a method to keep the activate() method short and easy to understand.
E.g. `exportButton.setOnAction(event -> doExportToJson()`

File: desktop/src/main/java/bisq/desktop/util/GUIUtil.java [hunk 5]
```diff
@@ -268,6 +275,21 @@ public static void importAccounts(User user, String fileName, Preferences prefer
         }
     }
 
+    public static void exportJSON(String fileName, JsonElement data, Stage stage) {
```
Comment (ManfredKarrer) [inline desktop/src/main/java/bisq/desktop/util/GUIUtil.java:275]: Is there a pretty print feature in the json framework? Would be nice to have, but if not no worry.

File: desktop/src/main/java/bisq/desktop/main/dao/governance/result/VoteResultView.java (hunk 4 above)
Comment (ManfredKarrer) [reply desktop/src/main/java/bisq/desktop/main/dao/governance/result/VoteResultView.java:198]: Could you add all fields (if missing, havve not checked all) from the domain objects which are currently not displayed in the UI.
We want to add popups for that anyway and for the json I think its good to have the complete domain data.

The CompensationProposal is missing `requestedBsq` and `bsqAddress`. As said the UI does not show all that yet but would be good to have that in the json.

Comment (ManfredKarrer) [reply desktop/src/main/java/bisq/desktop/main/dao/governance/result/VoteResultView.java:198]: For Proposal there are a few subclasses with diff. fields...

**Review by ManfredKarrer — CHANGES_REQUESTED** [review]
Thanks for the fast implementation! Looks basically very good, just a few Nits...

**Comment by ben-kaufman** [comment]
@ManfredKarrer Fixed all the issues above. Attaching here new screenshot and JSON.
screen shot 2019-03-06 at 23 40 37

[voteResultsHistory.json.zip](https://github.com/bisq-network/bisq/files/2938409/voteResultsHistory.json.zip)

**Review by ManfredKarrer — APPROVED** [review]
utACK

**Comment by ManfredKarrer** [comment]
Great thanks for the fast fixes! We can need devs like you! ;-)

**Comment by ManfredKarrer** [comment]
@ben-kaufman 
Thanks for adding the translations as well but unfortunately those will be overwritten at next release from Transifex files. We only should edit the default property file in english and the translations have to be done in Transifex. I will leave the commits, just wanted to let you know that the will be lost if no one will move them to Transifex.

**Pull request merged**